### PR TITLE
Consistently detect old driver when existing.Driver is unset

### LINKED ISSUE #7157: The existing "minikube" VM was created using the "hyperv" driver, and is incompatible with the "hyperv" driver.
Talk about a confusing error message!

```
$ ./out/minikube.exe  start --driver=hyperv --alsologtostderr
W0323 07:37:51.596645   23052 root.go:248] Error reading config file at C:\Users\spam1\.minikube\config\config.json: open C:\Users\spam1\.minikube\config\config.json: The system cannot find the file specified.
I0323 07:37:51.597646   23052 notify.go:125] Checking for updates...
I0323 07:37:51.721654   23052 start.go:254] hostinfo: {"hostname":"voimala","uptime":798533,"bootTime":1584175738,"procs":255,"os":"windows","platform":"Microsoft Windows 10 Pro","platformFamily":"Standalone Workstation","platformVersion":"10.0.19041 Build 19041","kernelVersion":"","virtualizationSystem":"","virtualizationRole":"","hostid":"f543d60d-b67a-46ae-9f82-beca49e73234"}
W0323 07:37:51.721654   23052 start.go:262] gopshost.Virtualization returned error: not implemented yet
* minikube v1.9.0-beta.2 on Microsoft Windows 10 Pro 10.0.19041 Build 19041
! The existing "minikube" VM was created using the "hyperv" driver, and is incompatible with the "hyperv" driver.
  - To proceed, either:

1) Delete the existing "minikube" cluster using: 'minikube delete'

* or *

2) Start the existing "minikube" cluster using: 'minikube start --driver=hyperv'

X Exiting.
```

### PR DESCRIPTION
**Pull request by tstromberg**
I had an issue upgrading a v1.6.0 created config with HEAD, because the "Driver" field was empty in the old configuration.

Fixes #7157

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -519,46 +520,58 @@ func selectDriver(existing *config.ClusterConfig) registry.DriverState {
 	return pick
 }
 
+// hostDriver returns the true driver used without relying on config fields
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go]: the comment is not clear to me, what is a True driver, 
maybe elaborate more on that.

Comment (sharifelgamal) [reply cmd/minikube/cmd/start.go]: it just means he's grabbing the driver from the machine itself rather than the config

Comment (medyagh) [reply cmd/minikube/cmd/start.go]: I rather comments that everyone can undrestand, including new contributors

**Pull request merged**